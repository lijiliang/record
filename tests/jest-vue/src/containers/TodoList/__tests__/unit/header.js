import { shallowMount } from '@vue/test-utils'
import Header from '../../components/Header.vue'
import { findTestWrapper } from '../../../../utils/testUtils'

it('Header 样式发生改变，做提示', () => {
  const wrapper = shallowMount(Header)
  expect(wrapper).toMatchSnapshot()
})

it('Header 包含 input 框', () => {
  const wrapper = shallowMount(Header)
  // const input = wrapper.find('[data-test="input"]') // 找到页面上的input框
  const input = findTestWrapper(wrapper, 'input')
  expect(input.exists()).toBe(true)
})

it('Header 包含 input 框初始内容为空', () => {
  const wrapper = shallowMount(Header)
  const inputValue = wrapper.vm.$data.inputValue
  expect(inputValue).toBe('')
})

it('Header 包含 input 发生变化，数据应该跟着变', () => {
  const wrapper = shallowMount(Header)
  const input = wrapper.find('[data-test="input"]') // 找到页面上的input框
  input.setValue('Benson')
  const inputValue = wrapper.vm.$data.inputValue
  expect(inputValue).toBe('Benson')
})

it('Header 包含 input 框输入回车，无内容时，无反应', () => {
  const wrapper = shallowMount(Header)
  const input = wrapper.find('[data-test="input"]') // 找到页面上的input框
  input.setValue('')
  input.trigger('keyup.enter') // 模拟用户输入回车
  expect(wrapper.emitted().add).toBeFalsy() // 没内容时，点回车时不应该向外面触发一个add事件
})

it('Header 包含 input 框输入回车，有内容时，向外触发事件，同时清空ipnutValue', () => {
  const wrapper = shallowMount(Header)
  const input = wrapper.find('[data-test="input"]') // 找到页面上的input框
  input.setValue('Benson')
  input.trigger('keyup.enter') // 模拟用户输入回车
  expect(wrapper.emitted().add).toBeTruthy() // 向外面触发一个add事件
  expect(wrapper.vm.inputValue).toBe('') // 清空inputValue
})
