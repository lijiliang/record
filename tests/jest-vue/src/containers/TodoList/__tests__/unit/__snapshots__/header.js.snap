// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 样式发生改变，做提示 1`] = `
<header>
  <section>
    <form action="javascript:postaction()" id="form"><label for="title">ToDoList</label> <input type="text" id="title" name="title" placeholder="添加ToDo" required="required" autocomplete="off" data-test="input"></form>
  </section>
</header>
`;
